File: tests/horners.py
def _horners(x, k, field, coeff, secret):
    ''' Evaluate the polynomial with Horner's method '''
    res = coeff[0]
    for i in range(1, k):
        res = ((res * x) % field + coeff[i]) % field
    return res

x = 0
k = 3
field = 257
coeff = [23,1,56,3]
secret = 245;
print(_horners(x,k,field,coeff,secret))

File: tests/horners.js
let mod = require('../number.js').mod;

function horners(x, k, field, poly, secret) {
        let result = secret;
        // evaluate polynomial with horner's method
        for(let i = 1; i < k; i++)
                        result = mod(mod(result * x, field) + poly[i], field);
        return result;
}

let x = 0;
let k = 3;
let field = 257;
let poly = [23,1,56,3];
let secret = 245;
console.log(horners(x,k,field,poly,secret));
